filters/builtin/timeout.go

```diff
@@ -17,24 +17,8 @@ func NewBackendTimeout() filters.Spec {
 func (*timeout) Name() string { return BackendTimeoutName }
 
 func (*timeout) CreateFilter(args []interface{}) (filters.Filter, error) {
-	if len(args) != 1 {
-		return nil, filters.ErrInvalidFilterParameters
-	}
-
-	var tf timeout
-	switch v := args[0].(type) {
-	case string:
-		d, err := time.ParseDuration(v)
-		if err != nil {
-			return nil, err
-		}
-		tf.timeout = d
-	case time.Duration:
-		tf.timeout = v
-	default:
-		return nil, filters.ErrInvalidFilterParameters
-	}
-	return &tf, nil
+	a := filters.Args(args)
+	return &timeout{a.Duration()}, a.Err()
```

Comment:
Expects single valid duration argument as parsable string or decimal number of milliseconds in any other case (no arguments, several arguments, invalid or negative duration string) `a.Err()` is not null and contains detailed error message like:
```
$ bin/skipper -inline-routes='* -> backendTimeout() -> <shunt>' 2>&1 | grep error
[APP]time="2021-07-10T23:22:36+02:00" level=error msg="failed to process route (): failed to create backendTimeout filter: expects 1 argument"

$ bin/skipper -inline-routes='* -> backendTimeout("xyz") -> <shunt>' 2>&1 | grep error
[APP]time="2021-07-10T23:22:54+02:00" level=error msg="failed to process route (): failed to create backendTimeout filter: time: invalid duration \"xyz\""

$ bin/skipper -inline-routes='* -> backendTimeout("1s", "x") -> <shunt>' 2>&1 | grep error
[APP]time="2021-07-10T23:23:11+02:00" level=error msg="failed to process route (): failed to create backendTimeout filter: expects 1 argument"

$ bin/skipper -inline-routes='* -> backendTimeout("-10s") -> <shunt>' 2>&1 | grep error
[APP]time="2021-07-10T23:23:20+02:00" level=error msg="failed to process route (): failed to create backendTimeout filter: duration -10s is negative"

$ bin/skipper -inline-routes='* -> backendTimeout("x", "y") -> <shunt>' 2>&1 | grep error
[APP]time="2021-07-10T23:24:54+02:00" level=error msg="failed to process route (): failed to create backendTimeout filter: expects 1 argument, time: invalid duration \"x\""
```

Note that constructor always returns `&timeout{}` value even in case of error - this should not be a big problem as it is common to check error first and throw away the value if error is not nil.